### ISSUE TITLE
diophantine solver not working well

### ISSUE THREAD

**Issue by dharasty** [sympy/sympy#11959]
I can readily find examples of the Diophantine solver missing solutions.

I'm working on a family of problems in the general binary quadratic equation form, but all my cases are the "elliptical case", that is B^2 - 4AC < 0.

Here is ONE set of coefficients that shows sympy is missing values: A=10, B=0, C=10, D=-6, E=-14, F=-52548.  (I can generate MANY more if you'd like; let me know if you need more test cases.)  

```
from sympy import sympify, symbols
from sympy.solvers.diophantine import diophantine

# A=10, B=0, C=10, D=-6, E=-14, F=-52548
eqn = sympify("10*x**2 - 6*x + 10*y**2 - 14*y - 52548")
print "eqn:", eqn

sympy_solution = diophantine(eqn)
print "sympy_solution:", sympy_solution
#set([(-70, -17), (72, -10), (54, -48), (-72, 6), (65, -32), (-59, -41), (-68, 25), (-61, -38)])

# calculated from https://www.alpertron.com.ar/QUAD.HTM
full_solution = set([
    (11, -71), (33, -64), (49, -53), (54, -48), 
    (-59, -41), (-61, -38), (65, -32), (-70, -17), 
    (72, -10), (-72, 6), (-68, 25), (42, 60), 
    (39, 62), (-24, 69), (18, 71), (-5, 73)
    ])

x, y = symbols('x y')
# prove that these are all solutions
for xval, yval in full_solution:
    val = eqn.subs({x: xval, y: yval}).simplify()
    if val != 0: raise "Oops..."

print "full_solution: ", full_solution

assert sympy_solution.issubset(full_solution), "Oops..."

missing = full_solution - sympy_solution # set difference
print "missing:       ", missing
# set([(-5, 73), (49, -53), (18, 71), (39, 62), (33, -64), (11, -71), (-24, 69), (42, 60)])
print "the sympy solutions is missing: %d values" % (len(missing))

```
Notes:
- curiously, another user just posted Diophantine failures today, too.  Was there a code update that broke something?
- [this page](https://www.alpertron.com.ar/QUAD.HTM) correctly finds all the solutions I expect, and has an excellent general description on [how to solve](https://www.alpertron.com.ar/METHODS.HTM) the general binary quadratic equation.

**Comment by pranjaltale16** [comment]
@dharasty could you provide me some more cases i have made some changes would love to test on more examples to comfirm things.

**Comment by dharasty** [comment]
OK, here are the test cases and code to run the test cases:
[diophantine_checker.py.txt](https://github.com/sympy/sympy/files/669167/diophantine_checker.py.txt)
[diophantine_tests.py.txt](https://github.com/sympy/sympy/files/669169/diophantine_tests.py.txt)
Both are python files, so remove ".txt" from the files before running.

Then run "diophantine_checker".  This imports all the tests from "diophantine_tests", which is a list of dictionaries that look like this:
```  
   {
   'eqn': '10*x**2 + 0*x*y + 10*y**2 + -6*x + -14*y + -4132', 
   'solns': set([(-17, -10), (11, 18), (-20, 2), (-1, 21)]), 
   'num':  4, 'p': 5, 'q': 99, 's': 23 
   },
```

In `diophantine_checker` I read the `eqn` string, call `sympify()` on that, find sympy's `diophantine()` results, then compare it to the `solns` set in my test vector.  `num` is the number of solutions in my `solns` set, and `p`, `q`, and `s` are the integers I used to generate the test.  (Too complicated to go in to all the details, though I can share how the diophantine coefficients A, B, C, D, E, F are related to these, if necessary.)

Caveats:
- I'm running Windows 10, Python 2.7, Sympy 1.0
- All of my test cases are "elliptical" diophantine equations.
- My results are based on brute force search over all integer (x, y) values -- this is tractable because I know a priori the radius of my ellipses.
- I check my results that they DO indeed solve the equation.
- I'm pretty confident that I've found ALL solutions to my equations, but can't be 100% sure.  Getting our test results to agree would be a confirmation that both of ours is working.
- In most cases, the current Sympy diophantine solver is missing about half of the solutions.


My results running these (with Sympy 1.0) is:
```
Results:
  test cases     : 162
  tests passed   : 16
  tests failed   : 146
SOME TESTS FAILED
```

**Comment by pranjaltale16** [comment]
@dharasty 
I tested your cases with my code and i got all cases(162 cases) passed. 
While testing with (Sympy1.0) I got 25 passed and 137 Failed.

**Comment by dharasty** [comment]
@pranjaltale16 -- glad to hear of your improvements!
So.... when will your code be available?

**Comment by pranjaltale16** [comment]
@dharasty  i have already made a pull request #11961. 
As soon as any Team Member will review it and merge., corresponding corrections will be available.

**Comment by smichr** [comment]
solved in #11989
